### PR TITLE
Dask on Ray version documentation update

### PR DESCRIPTION
**Pull request by rkooo567**
## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by rkooo567** [comment]
cc @jiaodong

**Review by jiaodong — APPROVED** [review]
unblocking since this looks like previous version update on docs, and i will make a new PR following the same pattern for 1.5.0 cut.

File: doc/source/dask-on-ray.rst [hunk 1]
```diff
@@ -22,7 +22,18 @@ specify any callable as the scheduler that you would like Dask to use to execute
 workload. Using the Dask-on-Ray scheduler, the entire Dask ecosystem can be executed on top of Ray.
 
 .. note::
-  As of Ray 1.4.1, Dask versions 2021.6.1 or lower are supported.
+  We always ensure the latest Dask versions are compatible to the Ray nightly.
```
Comment (clarkzinzow) [inline doc/source/dask-on-ray.rst:25]: Nit:
```suggestion
  We always ensure that the latest Dask versions are compatible with Ray nightly.
```

**Review by clarkzinzow — APPROVED** [review]

**Comment by rkooo567** [comment]
The table format is wrong now. Let me fix it!

**Comment by rkooo567** [comment]
@jiaodong can you cherry-pick this PR after I merge it? Also, did you run the dask on ray compatibility test?

**Comment by rkooo567** [comment]
@clarkzinzow also it'll be great if you can have a final look on the PR!

**Review by clarkzinzow — APPROVED** [review]
LGTM!

**Review by jiaodong — APPROVED** [review]

**Comment by rkooo567** [comment]
FAiled tests should be unrelated

**Pull request merged**